operator/pkg/lib/sidecars/restart/restart_test.go

```diff
@@ -0,0 +1,255 @@
+
+		obj := appsv1.Deployment{}
+		err = c.Get(context.TODO(), types.NamespacedName{Namespace: "test-ns", Name: "owner"}, &obj)
+		require.NoError(t, err)
+
+		require.NotEmpty(t, "owner", obj.Annotations[annotationName])
+	})
+
+	t.Run("should rollout restart DaemonSet if the pod is owned by one", func(t *testing.T) {
+		// given
+		c := fakeClient(t, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "owner", Namespace: "test-ns"}})
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				podFixture("p1", "test-ns", "DaemonSet", "owner"),
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.Empty(t, warnings)
+
+		obj := appsv1.DaemonSet{}
+		err = c.Get(context.TODO(), types.NamespacedName{Namespace: "test-ns", Name: "owner"}, &obj)
+		require.NoError(t, err)
+
+		require.NotEmpty(t, "owner", obj.Annotations[annotationName])
+	})
+
+	t.Run("should delete a pod belonging to a ReplicaSet with no owner", func(t *testing.T) {
+		// given
+		pod := podFixture("p1", "test-ns", "ReplicaSet", "owner")
+		c := fakeClient(t, &appsv1.ReplicaSet{ObjectMeta: metav1.ObjectMeta{Name: "owner", Namespace: "test-ns"}}, &pod)
+
+		podList := v1.PodList{
+			Items: []v1.Pod{pod},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.Empty(t, warnings)
+
+		obj := v1.Pod{}
+		err = c.Get(context.TODO(), types.NamespacedName{Namespace: "test-ns", Name: "p1"}, &obj)
+
+		require.Error(t, err)
+		require.True(t, k8serrors.IsNotFound(err))
+	})
+
+	t.Run("should rollout restart StatefulSet if the pod is owned by one", func(t *testing.T) {
+		// given
+		pod := podFixture("p1", "test-ns", "StatefulSet", "owner")
+
+		c := fakeClient(t, &appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Name: "owner", Namespace: "test-ns"}}, &pod)
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				pod,
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.Empty(t, warnings)
+
+		obj := appsv1.StatefulSet{}
+		err = c.Get(context.TODO(), types.NamespacedName{Namespace: "test-ns", Name: "owner"}, &obj)
+		require.NoError(t, err)
+
+		require.NotEmpty(t, "owner", obj.Annotations[annotationName])
+	})
+
+	t.Run("should return a warning when Pod is owned by a ReplicaSet that is not found", func(t *testing.T) {
+		// given
+		pod := podFixture("p1", "test-ns", "ReplicaSet", "podOwner")
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				pod,
+			},
+		}
+
+		c := fakeClient(t, &pod)
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, warnings)
+
+		pods := v1.PodList{}
+		err = c.List(context.TODO(), &pods)
+
+		require.NoError(t, err)
+		require.NotEmpty(t, pods.Items)
+	})
+
+	t.Run("should not delete pod when it is owned by a ReplicaSet that is found", func(t *testing.T) {
```

Comment:
I think to align with the other tests this can be changed to `should rollout restart ReplicaSet if the pod is owned by one`. This also means that the Pod is not deleted.

Reply:
Discussion outcome: Add the `should rollout restart ReplicaSet if the pod is owned by one` as a new separate test and keep the `should not delete pod when it is owned by a ReplicaSet that is found` test.

Reply:
https://github.com/kyma-project/istio/pull/37/commits/def272fdf7cee61fd9bb8ad2ac6616ec3acb9228